resources/run_adl.sh

```diff
@@ -0,0 +1,195 @@
+  result=$?
+  
+  if [[ result -eq 0 ]]; then
+    return 0
+  else 
+    return 1
+  fi
+}
+
+check_port() {
+  netstat -van 2>/dev/null | grep LISTEN | grep $1 >/dev/null
+  result=$?
+  
+  if [[ result -eq 0 ]]; then
+    return 0
+  else 
+    return 1
+  fi
+}
+
+check_mongod() {
+  check_procname $MONGOD
+  process_check_result=$?
+  check_port $MONGOD_PORT
+  port_check_result=$?
+
+  if [[ $process_check_result -eq 0 ]] && [[ $port_check_result -eq 0 ]]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+check_mongohoused() {
+  check_procname $MONGOHOUSED
+  process_check_result=$?
+  check_port $MONGOHOUSED_PORT
+  port_check_result=$?
+
+  if [[ $process_check_result -eq 0 ]] && [[ $port_check_result -eq 0 ]]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+# check if mac or linux
+if [ $OS = "Linux" ]; then
+  distro=$(awk -F= '/^NAME/{print $2}' /etc/os-release)
+  if [ "$distro" = "\"Red Hat Enterprise Linux\"" ] ||
+[ "$distro" = "\"Red Hat Enterprise Linux Server\"" ]; then
+    export VARIANT=rhel7
+    MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_REDHAT
+  elif [ "$distro" = "\"Ubuntu\"" ]; then
+    export VARIANT=ubuntu1804  
+    MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_UBUNTU
+  else
+    echo ${distro} not supported
+    exit 1
+  fi
+elif [ $OS = "Darwin" ]; then
+  MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_MAC
+else
+  echo $(uname) not supported
+  exit 1
+fi
+
+check_mongod
+if [[ $? -ne 0 ]]; then
+  if [ $ARG = $START ]; then
+    # Install and start mongod
+    if [ $OS = "Linux" ]; then
+      curl -O $MONGO_DOWNLOAD_BASE/linux/$MONGO_DOWNLOAD_LINK
+    else
+      curl -O $MONGO_DOWNLOAD_BASE/osx/$MONGO_DOWNLOAD_LINK
+    fi
+
+    tar zxvf $MONGO_DOWNLOAD_LINK
+
+    mkdir -p $MONGO_DB_PATH
+    mkdir -p $LOGS_PATH
+    mkdir -p $TMP_DIR
+
+    # Note: ADL has a storage.json file that generates configs for us.
+    # The mongodb source is on port $MONGOD_PORT so we use that here.
+    ${MONGO_DOWNLOAD_LINK:0:$((${#MONGO_DOWNLOAD_LINK} - 4))}/bin/mongod --port $MONGOD_PORT --dbpath $MONGO_DB_PATH \
+      --logpath $LOGS_PATH/mongodb_test.log --pidfilepath $TMP_DIR/${MONGOD}.pid --fork
+  fi
+else
+  if [ $ARG = $STOP ]; then
+    MONGOD_PID=$(< $TMP_DIR/${MONGOD}.pid)
+    echo "Stopping $MONGOD, pid $MONGOD_PID"
+    kill "$(< $TMP_DIR/${MONGOD}.pid)"
+  fi
+fi
+
+check_mongohoused
+if [[ $? -ne 0 ]]; then
+  if [ $ARG = $START ]; then
+    # Install and start mongohoused
+    git config --global url.git@github.com:.insteadOf https://github.com/
+    # Clone the mongohouse repo
+    if [ ! -d "$MONGOHOUSE_DIR" ]; then
+      git clone $MONGOHOUSE_URI
+    fi
+    cd $MONGOHOUSE_DIR
+    git pull $MONGOHOUSE_URI
+
+    go mod download
+
+    # Download external dependencies
+    go run cmd/buildscript/build.go tools:download:mqlrun
+    go run cmd/buildscript/build.go tools:download:mongosql
+
+    # Set relevant environment variables
+    export GOPRIVATE=github.com/10gen
```

Comment:
I'm surprised you don't need to do this before doing the `go mod download` since that command is one that _should_ rely on this. For example, when it attempts to download the `10gen/mongoast` dependency I'm surprised it doesn't complain. If this is working, I suppose there is no need to change it.

Reply:
I looked into this a bit and I'm finding that the command that helps `go mod download` succeed (tested on a new ubuntu1804 host) is:
```git config --global url.git@github.com:.insteadOf https://github.com/```
Otherwise it complains with the error:
```
$ GOPRIVATE=github.com/10gen go mod download
go: github.com/10gen/bsonio@v0.0.0-20211202155813-2e4634d25cd1: invalid version: git fetch -f origin refs/heads/*:refs/heads/* refs/tags/*:refs/tags/* in /home/ubuntu/go/pkg/mod/cache/vcs/3a9008e6621e88540dcd7b013a52c240ba3677268aedf02ee2f397893fbb4f73: exit status 128:
	fatal: could not read Username for 'https://github.com': terminal prompts disabled
```
I'll move the export GOPRIVATE above the `go mod download` command just in case.

Reply:
oh that makes so much sense -- thanks for looking into that and figuring out what was happening